app/models/test.rb

```diff
@@ -4,11 +4,23 @@ class Test < ApplicationRecord
   has_many :users, through: :results
   belongs_to :category
   belongs_to :authored_tests, class_name: 'User', foreign_key: 'user_id'
+  validates :title, presence: true,
+                    uniqueness: { only_integer: true, 
```

Comment:
[в документации](https://guides.rubyonrails.org/active_record_validations.html#uniqueness) к хелперу `uniqueness` сказано следующее:
> It does not create a uniqueness constraint in the database, so it may happen that two different database connections create two records with the same value for a column that you intend to be unique. To avoid that, you must create a unique index on that column in your database.

другими словами, чтобы эта была настоящая уникальность (не только на уровне приложения, но и на уровне базы), нужно в базе создать уникальный индекс

примерно таким образом:

```ruby
class AddIndexToTests < ActiveRecord::Migration[5.2]
  def change
    add_index :tests, [:title, :level], unique: true
  end
end
```